src/lua/fiber_cond.c

```diff
@@ -52,32 +50,53 @@ luaT_fiber_cond_new(struct lua_State *L)
 	return 1;
 }
 
-static inline struct fiber_cond *
-luaT_checkfibercond(struct lua_State *L, int index, const char *source)
+static struct fiber_cond *
+luaT_isfibercond(struct lua_State *L, int index)
 {
 	if (index > lua_gettop(L))
-		luaL_error(L, "usage: %s", source);
-	return (struct fiber_cond *)luaL_checkudata(L, index, cond_typename);
+		return NULL;
+	return luaL_checkudata(L, index, cond_typename);
+}
+
+struct fiber_cond *
+luaT_checkfibercond(struct lua_State *L, int index)
+{
+	struct fiber_cond *fiber_cond = luaT_isfibercond(L, index);
+	if (fiber_cond == NULL) {
+		luaL_error(L, "Invalid argument #%d"
+			   "(fiber.cond expected, got %s)", index,
+			   lua_typename(L, lua_type(L, index)));
+	}
+	return fiber_cond;
 }
 
 static int
 luaT_fiber_cond_gc(struct lua_State *L)
 {
-	fiber_cond_destroy(luaT_checkfibercond(L, 1, "cond:destroy()"));
+	struct fiber_cond *fiber_cond = luaT_isfibercond(L, 1);
+	if (fiber_cond == NULL)
+		luaL_error(L, "usage: %s", "cond:destroy()");
```

Comment:
%s isn't necessary here - you can just as well write:

```C
luaL_error(L, "usage: cond:destroy()");
```

Please fix here and below.

Reply:
fixed